docs/source/algorithms/MaskNonOverlappingBins-v1.rst

```diff
@@ -0,0 +1,40 @@
+
+.. algorithm::
+
+.. summary::
+
+.. relatedalgorithms::
+
+.. properties::
+
+Description
+-----------
+
+This algorithm masks the bins in ``InputWorkspace`` which lie in :math:`X` range that is not covered by ``ComparisonWorkspace``. The ``MaskPartiallyOverlapping`` flag affect the behavior with regards to bins which are partially covered by ``ComparisonWorkspace``.
+
+The algorithm currently applies the default masking weight to the bins which does not clear the data.
```

Comment:
Good documentation. What is the reason for not clearing the data. I'm interested in the decision behind this.

Reply:
Here I just used the default behavior and didn't bother adding something like `MaskWeight` property. I'm happy with the default as generally I think destroying the data underneath a mask should be a separate operation from masking.

Reply:
I agree.